runtime.md

```diff
@@ -67,28 +65,45 @@ This operation MUST return the state of a container as specified in the [State](
 In particular, the state MUST be serialized as JSON.
 
 
-### Start
+### Create
 
-`start <container-id> <path-to-bundle>`
+`create <container-id> <path-to-bundle>`
 
 This operation MUST generate an error if it is not provided a path to the bundle and the container ID to associate with the container.
-If the ID provided is not unique across all containers within the scope of the runtime, or is not valid in any other way, the implementation MUST generate an error.
-Using the data in `config.json`, that are in the bundle's directory, this operation MUST create a new container.
-This includes creating the relevant namespaces, resource limits, etc and configuring the appropriate capabilities for the container.
-A new process within the scope of the container MUST be created as specified by the `config.json` file otherwise an error MUST be generated.
+If the ID provided is not unique across all containers within the scope of the runtime, or is not valid in any other way, the implementation MUST generate an error and a new container MUST not be created.
+Using the data in [`config.json`](config.md), that is in the root of the bundle's directory, this operation MUST create a new container.
+This includes creating, or entering, the namespaces specified in the [`config.json`](config.md), resource limits, etc and configuring the appropriate capabilities for the container.
+If the `config.json` specifies that a PID namespace is to be created then one MUST be created, but the user-specified code within that namespace MUST NOT be created at this time.
+In some implementations this means that a temporary process is created in the PID namespace but it pauses until the `start` operation is invoked before replacing the process with the user-specified code.
+
+### Start
+`start <container-id>`
 
-Attempting to start an already running container MUST have no effect on the container and MUST generate an error.
+This operation MUST generate an error if it is not provided the container ID.
+This operation MUST create, and start, the user-specified code as specified by the [`process`](config.md#process-configuration) file otherwise an error MUST be generated and no process MUST be started.
+This process MUST be run in the PID namespace associated with the container.
 
-### Stop
+Attempting to start an already started container MUST have no effect on the container and MUST generate an error.
 
-`stop <container-id>`
+### Run
+`run <container-id>`
 
 This operation MUST generate an error if it is not provided the container ID.
+This operation MUST invoke the `create` operation, and if there are no errors, then invoke the `start` operation.
+The implementation MAY immediately execute `start` without any pause after `create` completes.
+For example if the implementation used a temporary process to implement `create`, it need not do this in the implementation of `run`.
+
+### Delete
```

Comment:
On Thu, Apr 14, 2016 at 01:58:20PM -0700, Vish Kannan wrote:
> > +### Delete
> 
> How do we handle graceful deletions? As of now the general norm is
> that of posting a `SIGTERM` followed by a `SIGKILL`.

Can we keep the delete discussion in [1,2]?  It seems orthogonal to a
create/start split.

[1]: https://github.com/opencontainers/runtime-spec/issues/356
[2]: https://groups.google.com/a/opencontainers.org/forum/#!msg/dev/BCt9tI9lwEE/9__nW3waBgAJ
     Subject: Clarify distinction between ‘stop’ and ‘delete’
     Date: Wed, 23 Mar 2016 11:49:35 -0700
     Message-ID: <20160323184935.GB23066@odin.tremily.us>